server/downloader.h

```diff
@@ -6,29 +6,23 @@
 #include <filesystem>
 
 class downloader {
-public:
-    downloader(std::string path, std::string server);
 
-    bool valid() const;
+private:
```

Comment:
please reorder the public and private functions
see: https://google.github.io/styleguide/cppguide.html#Declaration_Order

> Group similar declarations together, placing public parts earlier.
> 
> A class definition should usually start with a public: section, followed by protected:, then private:. Omit sections that would be empty.